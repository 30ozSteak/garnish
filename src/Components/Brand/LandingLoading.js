import React from "react";
import "../Brand/LandingLoading";
import "../Brand/LandingLoading.css";
import "../../Styles/colors.css";

const LandingLoading = () => {
  return (
    <div className="bigger-slot-box">
      <div className="slot-box">
        <h1 class="slot-machine">
          <span class="slot-value slot-value-initial slot-value-slide-out">
            news
          </span>
          <span class="slot-value slot-cycle">technology</span>
          <span class="slot-value slot-cycle">diversity</span>
          <span class="slot-value slot-cycle">meme</span>
          <span class="slot-value slot-cycle">culture</span>
          <span class="slot-value slot-cycle slot-value-final">garn</span>
          <span className="ish">ish</span>
        </h1>
        <p className="logo-subtitle">a trim of news for your day</p>
      </div>
    </div>
  );
};

export default LandingLoading;
