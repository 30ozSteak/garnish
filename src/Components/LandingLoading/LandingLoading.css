* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.bigger-slot-box {
  display: flex;
  justify-content: center;
}

.slot-box {
  margin-top: 18rem;
  margin-right: 7rem;
  width: 18rem;
}

.slot-machine {
  font-size: 40px;
  font-weight: 600;
  height: 90px;
  line-height: 90px;
  overflow: hidden;
  position: relative;
  text-align: right;
}

.slot {
  position: absolute;
  right: 90px;
  transform: translate(50px, -1000%);
}

.in {
  transform: translate(20px, 0);
}

.out {
  animation: slide-out 0.8s forwards;
}

.cycle {
  animation: slide-in-and-out 1.5s;
}

.slot-final {
  animation: slide-in 1.4s ease-in forwards;
}

.ish {
  border: 5px solid var(--GFmint);
  padding: 2px;
}

.logo-subtitle {
  animation: appear 7.5s linear;
  font-size: 0.8rem;
  font-weight: 200;
  margin-top: -15px;
  text-align: right;
}

@keyframes appear {
  0% {
    opacity: 0;
  }

  90% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@keyframes slide-in {
  10% {
    transform: translate(20px, -100%);
  }
  30% {
    transform: translate(20px, 0);
  }
  80% {
    transform: translate(20px, 0);
  }
  100% {
    transform: translate(20px, 0);
  }
}

@keyframes slide-out {
  10% {
    transform: translate(20px, 0%);
  }
  100% {
    transform: translate(20px, 105%);
  }
}

@keyframes slide-in-and-out {
  10% {
    transform: translate(20px, -100%);
  }
  30% {
    transform: translate(20px, 0);
  }
  60% {
    transform: translate(20px, 0);
  }
  70% {
    transform: translate(20px, 0);
  }
  100% {
    transform: translate(20px, 105%);
  }
}

.slot:nth-child(1),
.slot:nth-child(2) {
  animation-delay: 1.2s;
}
.slot:nth-child(3) {
  animation-delay: 2.2s;
}
.slot:nth-child(4) {
  animation-delay: 3.2s;
}
.slot:nth-child(5) {
  animation-delay: 4.2s;
}
.slot:nth-child(6) {
  animation-delay: 5.2s;
}
