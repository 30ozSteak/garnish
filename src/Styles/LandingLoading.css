* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.bigger-slot-box {
  display: flex;
  justify-content: center;
}

.slot-box {
  margin-top: 20rem;
  margin-right: 7rem;
  width: 18rem;
}

.slot-machine {
  font-size: 40px;
  font-weight: 600;
  height: 90px;
  line-height: 90px;
  overflow: hidden;
  position: relative;
  text-align: right;
}

.slot-value {
  position: absolute;
  right: 90px;
  transform: translate(50px, -1000%);
}

.slot-value-initial {
  transform: translate(20px, 0);
}

.slot-value-slide-out {
  animation: slide-out 0.5s ease-in forwards;
}

.slot-cycle {
  animation: slide-in-and-out 1.3s ease-in;
}

.slot-value-final {
  animation: slide-in 1.6s ease-in forwards;
}

.ish {
  border: 5px solid #68d8b8;
  padding: 2px;
}

.logo-subtitle {
  animation: appear 8.5s linear;
  font-size: 0.8rem;
  font-weight: 200;
  margin-top: -15px;
  text-align: right;
}

@keyframes appear {
  0% {
    opacity: 0;
  }

  90% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@keyframes slide-in {
  10% {
    transform: translate(20px, -100%);
  }
  30% {
    transform: translate(20px, 0);
  }
  80% {
    transform: translate(20px, 0);
  }
  100% {
    transform: translate(20px, 0);
  }
}

@keyframes slide-out {
  10% {
    transform: translate(20px, -15%);
  }
  100% {
    transform: translate(20px, 105%);
  }
}

@keyframes slide-in-and-out {
  10% {
    transform: translate(20px, -100%);
  }
  30% {
    transform: translate(20px, 0);
  }
  60% {
    transform: translate(20px, 0);
  }
  70% {
    transform: translate(20px, -15%);
  }
  100% {
    transform: translate(20px, 105%);
  }
}

.slot-value:nth-child(1),
.slot-value:nth-child(2) {
  animation-delay: 1.2s;
}
.slot-value:nth-child(3) {
  animation-delay: 2.2s;
}
.slot-value:nth-child(4) {
  animation-delay: 3.3s;
}
.slot-value:nth-child(5) {
  animation-delay: 4.3s;
}
.slot-value:nth-child(6) {
  animation-delay: 5.5s;
}
