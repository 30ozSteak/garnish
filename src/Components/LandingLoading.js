import React from "react";
import "../Styles/LandingLoading.css";

const LandingLoading = () => {
  return (
    <div className="slottest-box">
      <div className="slot-box">
        <h1 class="slot-machine">
          <span class="slot-value slot-value-initial slot-value-slide-out">
            news
          </span>
          <span class="slot-value slot-cycle">technology</span>
          <span class="slot-value slot-cycle">diversity</span>
          <span class="slot-value slot-cycle">meme</span>
          <span class="slot-value slot-cycle">culture</span>
          <span class="slot-value slot-cycle slot-value-final">garn</span>
          <span className="ish">ish</span>
        </h1>
        <p className="logo-subtitle">a trim of news for your day</p>
      </div>
    </div>
  );
};

export default LandingLoading;
