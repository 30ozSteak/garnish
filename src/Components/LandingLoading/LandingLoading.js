import React from "react";
import { NavLink } from "react-router-dom";

import "../LandingLoading/LandingLoading";
import "../LandingLoading/LandingLoading.css";
import "../../Styles/colors.css";

const LandingLoading = () => {
  return (
    <div className="bigger-slot-box">
      <div className="slot-box">
        <h1 className="slot-machine">
          <span className="slot in out">news</span>
          <span className="slot cycle">tech</span>
          <span className="slot cycle">diversity</span>
          <span className="slot cycle">meme</span>
          <span className="slot cycle">culture</span>
          <span className="slot cycle slot-final">garn</span>
          <span className="ish">ish</span>
        </h1>
        <p className="logo-subtitle">a trim of news for your day</p>
      </div>
      <NavLink to="/main">continue</NavLink>
    </div>
  );
};

export default LandingLoading;
