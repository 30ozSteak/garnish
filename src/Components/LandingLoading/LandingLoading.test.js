import React from "react";
import LandingLoading from "./LandingLoading";
import { shallow } from "enzyme";

describe("LandingLoading", () => {
  it("should render pretty neat landing animation that matches snapshot", () => {
    const wrapper = shallow(<LandingLoading />);
    expect(wrapper).toMatchSnapshot();
  });
});
