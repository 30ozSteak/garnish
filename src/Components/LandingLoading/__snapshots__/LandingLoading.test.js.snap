// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingLoading should render pretty neat landing animation that matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingLoading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="slot-box"
        >
          <h1
            className="slot-machine"
          >
            <span
              className="slot in out"
            >
              news
            </span>
            <span
              className="slot cycle"
            >
              technology
            </span>
            <span
              className="slot cycle"
            >
              diversity
            </span>
            <span
              className="slot cycle"
            >
              meme
            </span>
            <span
              className="slot cycle"
            >
              culture
            </span>
            <span
              className="slot cycle slot-final"
            >
              garn
            </span>
            <span
              className="ish"
            >
              ish
            </span>
          </h1>
          <p
            className="logo-subtitle"
          >
            a trim of news for your day
          </p>
        </div>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/main"
        >
          continue
        </NavLink>,
      ],
      "className": "bigger-slot-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="slot-machine"
            >
              <span
                className="slot in out"
              >
                news
              </span>
              <span
                className="slot cycle"
              >
                technology
              </span>
              <span
                className="slot cycle"
              >
                diversity
              </span>
              <span
                className="slot cycle"
              >
                meme
              </span>
              <span
                className="slot cycle"
              >
                culture
              </span>
              <span
                className="slot cycle slot-final"
              >
                garn
              </span>
              <span
                className="ish"
              >
                ish
              </span>
            </h1>,
            <p
              className="logo-subtitle"
            >
              a trim of news for your day
            </p>,
          ],
          "className": "slot-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="slot in out"
                >
                  news
                </span>,
                <span
                  className="slot cycle"
                >
                  technology
                </span>,
                <span
                  className="slot cycle"
                >
                  diversity
                </span>,
                <span
                  className="slot cycle"
                >
                  meme
                </span>,
                <span
                  className="slot cycle"
                >
                  culture
                </span>,
                <span
                  className="slot cycle slot-final"
                >
                  garn
                </span>,
                <span
                  className="ish"
                >
                  ish
                </span>,
              ],
              "className": "slot-machine",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "news",
                  "className": "slot in out",
                },
                "ref": null,
                "rendered": "news",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "technology",
                  "className": "slot cycle",
                },
                "ref": null,
                "rendered": "technology",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "diversity",
                  "className": "slot cycle",
                },
                "ref": null,
                "rendered": "diversity",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "meme",
                  "className": "slot cycle",
                },
                "ref": null,
                "rendered": "meme",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "culture",
                  "className": "slot cycle",
                },
                "ref": null,
                "rendered": "culture",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "garn",
                  "className": "slot cycle slot-final",
                },
                "ref": null,
                "rendered": "garn",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ish",
                  "className": "ish",
                },
                "ref": null,
                "rendered": "ish",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "a trim of news for your day",
              "className": "logo-subtitle",
            },
            "ref": null,
            "rendered": "a trim of news for your day",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "continue",
          "to": "/main",
        },
        "ref": null,
        "rendered": "continue",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="slot-box"
          >
            <h1
              className="slot-machine"
            >
              <span
                className="slot in out"
              >
                news
              </span>
              <span
                className="slot cycle"
              >
                technology
              </span>
              <span
                className="slot cycle"
              >
                diversity
              </span>
              <span
                className="slot cycle"
              >
                meme
              </span>
              <span
                className="slot cycle"
              >
                culture
              </span>
              <span
                className="slot cycle slot-final"
              >
                garn
              </span>
              <span
                className="ish"
              >
                ish
              </span>
            </h1>
            <p
              className="logo-subtitle"
            >
              a trim of news for your day
            </p>
          </div>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/main"
          >
            continue
          </NavLink>,
        ],
        "className": "bigger-slot-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="slot-machine"
              >
                <span
                  className="slot in out"
                >
                  news
                </span>
                <span
                  className="slot cycle"
                >
                  technology
                </span>
                <span
                  className="slot cycle"
                >
                  diversity
                </span>
                <span
                  className="slot cycle"
                >
                  meme
                </span>
                <span
                  className="slot cycle"
                >
                  culture
                </span>
                <span
                  className="slot cycle slot-final"
                >
                  garn
                </span>
                <span
                  className="ish"
                >
                  ish
                </span>
              </h1>,
              <p
                className="logo-subtitle"
              >
                a trim of news for your day
              </p>,
            ],
            "className": "slot-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="slot in out"
                  >
                    news
                  </span>,
                  <span
                    className="slot cycle"
                  >
                    technology
                  </span>,
                  <span
                    className="slot cycle"
                  >
                    diversity
                  </span>,
                  <span
                    className="slot cycle"
                  >
                    meme
                  </span>,
                  <span
                    className="slot cycle"
                  >
                    culture
                  </span>,
                  <span
                    className="slot cycle slot-final"
                  >
                    garn
                  </span>,
                  <span
                    className="ish"
                  >
                    ish
                  </span>,
                ],
                "className": "slot-machine",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "news",
                    "className": "slot in out",
                  },
                  "ref": null,
                  "rendered": "news",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "technology",
                    "className": "slot cycle",
                  },
                  "ref": null,
                  "rendered": "technology",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "diversity",
                    "className": "slot cycle",
                  },
                  "ref": null,
                  "rendered": "diversity",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "meme",
                    "className": "slot cycle",
                  },
                  "ref": null,
                  "rendered": "meme",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "culture",
                    "className": "slot cycle",
                  },
                  "ref": null,
                  "rendered": "culture",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "garn",
                    "className": "slot cycle slot-final",
                  },
                  "ref": null,
                  "rendered": "garn",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "ish",
                    "className": "ish",
                  },
                  "ref": null,
                  "rendered": "ish",
                  "type": "span",
                },
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "a trim of news for your day",
                "className": "logo-subtitle",
              },
              "ref": null,
              "rendered": "a trim of news for your day",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "continue",
            "to": "/main",
          },
          "ref": null,
          "rendered": "continue",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
